PdfScribe/Program.cs

```diff
@@ -138,13 +274,214 @@ static void Main(string[] args)
+        {
+            bool filenameRetrieved = false;
+            //outputFile = GetFreeRdpOutputFilename(_parentProcess);
+            if (!string.IsNullOrEmpty(outputFile))
+            {
+                filenameRetrieved = true;
+            }
+            else
+            {
+                switch (Properties.Settings.Default.AskUserForOutputFilename)
+                {
+                    case (true):
+                        using (SetOutputFilename dialogOwner = new SetOutputFilename())
+                        {
+                            dialogOwner.TopMost = true;
+                            dialogOwner.TopLevel = true;
+                            dialogOwner.Show(); // Form won't actually show - Application.Run() never called
+                                                // but having a topmost/toplevel owner lets us bring the SaveFileDialog to the front
+                            dialogOwner.BringToFront();
+                            using (SaveFileDialog pdfFilenameDialog = new SaveFileDialog())
+                            {
+                                pdfFilenameDialog.AddExtension = true;
+                                pdfFilenameDialog.AutoUpgradeEnabled = true;
+                                pdfFilenameDialog.CheckPathExists = true;
+                                pdfFilenameDialog.Filter = "pdf files (*.pdf)|*.pdf";
+                                pdfFilenameDialog.ShowHelp = false;
+                                pdfFilenameDialog.Title = Properties.Resources.ProductCaption + " - " + Properties.Resources.SetOutputFilename;
+                                pdfFilenameDialog.ValidateNames = true;
+                                if (!String.IsNullOrEmpty(Environment.GetEnvironmentVariable("REDMON_DOCNAME")))
+                                {
+                                    // Replace illegal characters with spaces
+                                    Regex regEx = new Regex(@"[\\/:""*?<>|]");
+                                    pdfFilenameDialog.FileName = regEx.Replace(Environment.GetEnvironmentVariable("REDMON_DOCNAME"), " ");
+                                }
+                                if (pdfFilenameDialog.ShowDialog(dialogOwner) == DialogResult.OK)
+                                {
+                                    outputFile = pdfFilenameDialog.FileName;
+                                    filenameRetrieved = true;
+                                }
+                            }
+                            dialogOwner.Close();
+                        }
+                        break;
+                    default:
+                        try
+                        {
+                            outputFile = GetOutputFilename();
+                            // Test if we can write to the destination
+                            using (FileStream newOutputFile = File.Create(outputFile))
+                            { }

```

Comment:
The using statement here seems redundant, if creation fails an exception would be thrown and newOutputFile would have nothing to dispose

Reply:
this is original codes, and this part of code will not access during real time, because the output file name always having values.

I think I will change the code in case future we allow user to choose their own output name.